packages/zowe-explorer/src/trees/ZoweTreeProvider.ts

```diff
@@ -302,6 +304,46 @@ export class ZoweTreeProvider<T extends IZoweTreeNode> {
         await Profiles.getInstance().createZoweSession(zoweFileProvider);
     }
 
+    /**
+     * Checks if a JWT token is used for authenticating the given profile name.
+     * If so, it will grab and decode the token to determine its expire date.
+     * If the token has expired, it will prompt the user to log in again.
+     *
+     * @param profileName The name of the profile to check the JWT token for
+     */
+    protected static async checkJwtTokenForProfile(profileName: string): Promise<void> {
+        const profInfo = await Profiles.getInstance().getProfileInfo();
+        const profAttrs = profInfo.getAllProfiles().find((prof) => prof.profName === profileName);
+        const knownProps = profInfo.mergeArgsForProfile(profAttrs, { getSecureVals: true }).knownArgs;
+        const tokenValueProp = knownProps.find((arg) => arg.argName === "tokenValue" && arg.argValue != null);
+
+        // Ignore if tokenValue is not a prop
+        if (tokenValueProp == null) {
+            return;
+        }
+
+        const tokenTypeProp = knownProps.find((arg) => arg.argName === "tokenType");
+        // Cannot decode LTPA tokens without private key
+        if (tokenTypeProp?.argValue == "LtpaToken2") {
+            return;
+        }
+
+        const fullToken = tokenValueProp.argValue.toString();
+        // JWT format: [header].[payload].[signature]
+        const tokenParts = fullToken.split(".");
+        try {
+            const payloadJson = JSON.parse(Buffer.from(tokenParts[1], "base64url").toString("utf8"));
+            if ("exp" in payloadJson) {
+                const expireDate = dayjs.unix(payloadJson.exp);
+                if (expireDate.isBefore(dayjs())) {
+                    await AuthUtils.promptUserForSsoLogin(profileName);
+                }
+            }
+        } catch (err) {
+            return;
```

Comment:
Curious if this should be moved to the SDKs 😅 
Or at least be made available to extenders 😅

Reply:
I like this idea, thanks for the suggestion! I moved this function implementation into Imperative's `ProfileInfo` class.
Here is the PR: https://github.com/zowe/zowe-cli/pull/2298

Once that's approved & merged, we can adopt it in this PR - this would mean that extenders also get access to it directly from the facility that they already use for profiles ^^